br/pkg/lightning/metric/metric.go

```diff
@@ -53,200 +53,272 @@ const (
+			prometheus.HistogramOpts{
+				Namespace:   namespace,
+				Subsystem:   subsystem,
+				Name:        "block_deliver_kv_pairs",
+				Help:        "number of KV pairs being sent out to importer",
+				ConstLabels: constLabels,
+				Buckets:     prometheus.ExponentialBuckets(1, 2, 10),
+			}, []string{"kind"}),
+	}
+}
+
+// RegisterTo registers all metrics to the given registry.
+func (c *Common) RegisterTo(r promutil.Registry) {
+	r.MustRegister(
+		c.ChunkCounter,
+		c.BytesCounter,
+		c.RowsCounter,
+		c.RowReadSecondsHistogram,
+		c.RowEncodeSecondsHistogram,
+		c.BlockDeliverSecondsHistogram,
+		c.BlockDeliverBytesHistogram,
+		c.BlockDeliverKVPairsHistogram,
+	)
+}
+
+// UnregisterFrom unregisters all metrics from the given registry.
+func (c *Common) UnregisterFrom(r promutil.Registry) {
+	r.Unregister(c.ChunkCounter)
+	r.Unregister(c.BytesCounter)
+	r.Unregister(c.RowsCounter)
+	r.Unregister(c.RowReadSecondsHistogram)
+	r.Unregister(c.RowEncodeSecondsHistogram)
+	r.Unregister(c.BlockDeliverSecondsHistogram)
+	r.Unregister(c.BlockDeliverBytesHistogram)
+	r.Unregister(c.BlockDeliverKVPairsHistogram)
+}
+
 // Metrics contains all metrics used by lightning.
 type Metrics struct {
 	ImporterEngineCounter                *prometheus.CounterVec
 	IdleWorkersGauge                     *prometheus.GaugeVec
 	KvEncoderCounter                     *prometheus.CounterVec
 	TableCounter                         *prometheus.CounterVec
 	ProcessedEngineCounter               *prometheus.CounterVec
-	ChunkCounter                         *prometheus.CounterVec
-	BytesCounter                         *prometheus.CounterVec
-	RowsCounter                          *prometheus.CounterVec
 	ImportSecondsHistogram               prometheus.Histogram
 	ChunkParserReadBlockSecondsHistogram prometheus.Histogram
 	ApplyWorkerSecondsHistogram          *prometheus.HistogramVec
-	RowReadSecondsHistogram              prometheus.Histogram
-	RowReadBytesHistogram                prometheus.Histogram
-	RowEncodeSecondsHistogram            prometheus.Histogram
 	RowKVDeliverSecondsHistogram         prometheus.Histogram
-	BlockDeliverSecondsHistogram         prometheus.Histogram
-	BlockDeliverBytesHistogram           *prometheus.HistogramVec
-	BlockDeliverKVPairsHistogram         *prometheus.HistogramVec
-	ChecksumSecondsHistogram             prometheus.Histogram
-	SSTSecondsHistogram                  *prometheus.HistogramVec
-	LocalStorageUsageBytesGauge          *prometheus.GaugeVec
-	ProgressGauge                        *prometheus.GaugeVec
+	// RowReadBytesHistogram records the number of bytes read from data source
+	// for a batch of rows.
+	// it's a little duplicate with RowsCounter of state = "restored".
+	RowReadBytesHistogram       prometheus.Histogram
+	ChecksumSecondsHistogram    prometheus.Histogram
+	SSTSecondsHistogram         *prometheus.HistogramVec
+	LocalStorageUsageBytesGauge *prometheus.GaugeVec
+	ProgressGauge               *prometheus.GaugeVec
```

Comment:
Why are some metrics not necessary to import into?

Reply:
some metric is added in lightning, but not used in dashboard. in this pr only add those needed in the panel

and some're not fit for import into, as we only import 1 table here, such as ChecksumSecondsHistogram